libdevcore/JsonUtils.cpp

```diff
@@ -31,3 +31,86 @@ void dev::validateFieldNames(json_spirit::mObject const& _obj, std::set<std::str
 			BOOST_THROW_EXCEPTION(UnknownField() << errinfo_comment(comment));
 		}
 }
+
+std::string dev::jsonTypeAsString(json_spirit::Value_type _type)
+{
+    switch (_type)
+    {
+    case json_spirit::obj_type:
+        return "json Object";
+    case json_spirit::array_type:
+        return "json Array";
+    case json_spirit::str_type:
+        return "json String";
+    case json_spirit::bool_type:
+        return "json Bool";
+    case json_spirit::int_type:
+        return "json Int";
+    case json_spirit::real_type:
+        return "json Real";
+    case json_spirit::null_type:
+        return "json Null";
+    default:
+        return "json n/a";
+    }
+}
+
+void dev::requireJsonFields(json_spirit::mObject const& _o, std::string const& _config,
+    std::map<std::string, jsonType> const& _validationMap)
+{
+    // check for unexpected fiedls
+    for (auto const& field : _o)
+    {
+        if (!_validationMap.count(field.first))
+        {
+            std::string const comment =
+                "Unexpected field '" + field.first + "' in config: " + _config;
+            std::cerr << comment << "\n"
+                      << json_spirit::write_string((json_spirit::mValue)_o, true) << "\n";
+            BOOST_THROW_EXCEPTION(UnknownField() << errinfo_comment(comment));
+        }
+    }
+
+    // check field types with validation map
+    for (auto const vmap : _validationMap)
```

Comment:
I'd maybe rename `vmap` to something more clearly expressing what it is, because it's difficult to follow what all `vmap.second.first`, `vmap.second.second` etc. are.

Or maybe as first lines of the loop body do
```
auto const& fieldName = vmap.first;
auto const& fieldTypesAndPresence = vmap.second;
```

Reply:
git does not show. but this peace is updated with user friendly names